tools/wasm-sourcemap.py

```diff
@@ -263,7 +263,7 @@ def build_sourcemap(entries, code_section_offset, prefixes, collect_sources):
     if column == 0:
       column = 1
     address = entry['address'] + code_section_offset
-    file_name = entry['file']
+    file_name = os.path.relpath(entry['file'])
```

Comment:
I'm not sure this is correct: it uses a path relative to the current directory, but browsers / spec expect `sources` to be relative to the source map file.

For example, consider case like following:
```bash
$ emcc ./temp.c -g4 -o dist/temp.wasm
```

Currently this produces a source map `dist/temp.wasm.map` with following contents:
```js
{"version":3,"sources":["./temp.c"],...
```

But this is incorrect, because `temp.c` is actually in a folder up and needs to be referenced as `../temp.c`.

Reply:
In this case wouldn't the user need to set up `--source-map-base` to fix that? But I've never really understood any of this stuff, hopefully you do...

Reply:
So `--source-map-base` helps browser to find a source map from WebAssembly module.

But once browser found that source map, it now needs to find source files. There are three potential ways for it to do that:
1. `sources` are specified as absolute URLs (not great for local files, but can work well enough for public libs).
2. `sources` are specified as relative URLs to the source map itself.
3. Source map contains `sourceRoot`, and then all `sources` are relative to that instead (could be used in our case as an alternative, but it's a bit more complicated IMO).

Reply:
Actually, maybe you're right, and in this case `--source-map-base` might be enough, at least if `.wasm.map` is always emitted alongside `.wasm`. Hmm...

Reply:
No, wait, scratch that, because relative URL still has to be relative to the source map itself. So the comment before still stands :)

Reply:
What browser is basically doing:
1) Extract `sourceMappingURL` from the WebAssembly module (this one is `{--source-map-base}/out.wasm.map`).
2) Load that source map.
3) If that source map contains `sourceRoot`, prepend it to each item in `sources`.
4) Resolve each item in `sources` relative to `sourceMappingURL`.

So we need each item in `sources` to be relative to `out.wasm.map`, no matter where it is.

Reply:
Thanks! I think I see a little better. I'm still confused about how that `sourceMappingURL` option is used in two places for seemingly different purposes, though?

Reply:
In both places it's used as "URL of the .wasm.map". First, to find the map, and later to resolve other files it references, but semantic meaning is the same.

It's similar to how most module systems work, too: if A imports B, and B imports C, then first B is found from A, and then reference to C is resolve relatively to B.